esrally/rally.py

```diff
@@ -144,6 +154,28 @@ def add_track_source(subparser):
         help="Limit the number of search results for recent races (default: 10).",
         default=10,
     )
+    list_parser.add_argument(
```

Comment:
With the way things are structured right now, all `list` commands will accept these new race-specific arguments, but can't do anything useful with them. It's not a huge deal, but seeing them in the `--help` output for something like `esrally list tracks --help` could add some confusion.

That said, there's already precedent for this. `--limit` only really applies to races, but you can pass it to any other `list` subcommand.

Could you perhaps change the help text for the new arguments to indicate that they're intended to be used for races only? Something like `Show only races for this track` in this case, for example.